Automattic/wp-calypso
# Remove useHappychat hook

Part of efforts to sunset Happy Chat and remove all references: pdm0RZ-BU-p2.

## Proposed Changes

* Remove `useHappychat` hook and replace its usages with hooks from https://github.com/Automattic/wp-calypso/pull/77499.
* Introduce a new `ChatButton` that is a much simplified and refactored version of `HappychatButton`. I'll replace all other uses of the `HappychatButton` component and remove it fully in a follow-up.

## Testing Instructions

- Go to your purchases in `/me/purchases`.
- Select a paid plan. Click `Cancel plan` and then `Cancel subscription`. This should load the pre-cancelation flow.
- Depending on your chat eligibility (and its availability), you should see a nudge in upper right corner:

<img width="1496" alt="Screenshot 2023-06-14 at 10 11 30" src="https://github.com/Automattic/wp-calypso/assets/3392497/97b716e7-a714-4c3c-9345-6f81db0389dc">

Do the same for a custom domain. You'll need to click "Remove domain" first which will disable autorenew for it. At which point you'll get the option to "Cancel domain". Once that's clicked, you'll get this dialog with a nudge to chat:
<img width="1363" alt="Screenshot 2023-06-13 at 14 58 11" src="https://github.com/Automattic/wp-calypso/assets/3392497/75466a96-7cd6-4162-a391-4cb7a8db2905">

- Make sure that the main support flow in Help Center still works.
